sql/src/main/java/io/crate/analyze/MultiSourceSelect.java

```diff
@@ -118,24 +93,52 @@ public QuerySpec querySpec() {
         return querySpec;
     }
 
-    private RelationSplitter initSources(Map<QualifiedName, AnalyzedRelation> sources) {
-
-        RelationSplitter splitter = RelationSplitter.process(querySpec, sources.values());
-        for (Map.Entry<QualifiedName, AnalyzedRelation> entry : sources.entrySet()) {
+    public void initializeSources() {
+        splitter.process();
+        for (Map.Entry<QualifiedName, AnalyzedRelation> entry : originalSources.entrySet()) {
             QuerySpec spec = splitter.getSpec(entry.getValue());
             assert spec != null;
             Source source = new Source(entry.getValue(), spec);
-            this.sources.put(entry.getKey(), source);
+            sources.put(entry.getKey(), source);
         }
-        return splitter;
     }
 
     /**
      * Returns an orderBy containing only orderings which are not already pushed down to sources. The optional is only
-     * pressent if there are orderBySymbols left. The existing orderBy is returned if it
+     * present if there are orderBySymbols left. The existing orderBy is returned if it
      * there are no changes.
      */
     public Optional<OrderBy> remainingOrderBy() {
         return splitter.remainingOrderBy();
     }
+
+    public Map<QualifiedName, AnalyzedRelation> originalSources() {
```

Comment:
why do we need originalsources? actually this would allow to operate on the original sources after the split, which does not make sense, right?